core/src/main/java/org/elasticsearch/common/lucene/search/Queries.java

```diff
@@ -179,8 +179,6 @@ public static int calculateMinShouldMatch(int optionalClauseCount, String spec)
             result = calc < 0 ? result + calc : calc;
         }
 
-        return (optionalClauseCount < result ?
-                optionalClauseCount : (result < 0 ? 0 : result));
-
+        return result < 0 ? 0 : result;
```

Comment:
It's ok to return 'result' directly (the value is re-checked in applyMinimumShouldMatch) except for ExtendedCommonTermsQuery which uses calculateMinShouldMatch directly without checking the returned result. I can modify ExtendedCommonTermsQuery if you think that it's clearer this way.